pkg/odo/cli/service/create.go

```diff
@@ -237,76 +273,99 @@ func (o *ServiceCreateOptions) Validate() (err error) {
-				// k8s does't have it installed by default but OCP does
 				return err
 			}
 
-			var almExamples []map[string]interface{}
-			val, ok := csv.Annotations["alm-examples"]
-			if ok {
-				err = json.Unmarshal([]byte(val), &almExamples)
-				if err != nil {
-					return errors.Wrap(err, "unable to unmarshal alm-examples")
+			if o.ServiceName != "" && !o.DryRun {
+				// First check if service with provided name already exists
+				svcFullName := strings.Join([]string{o.CustomResource, o.ServiceName}, "/")
+				exists, _ := svc.OperatorSvcExists(o.KClient, svcFullName)
+				if exists {
+					return fmt.Errorf("Service %q already exists. Please provide a different name or delete the existing service first.", svcFullName)
 				}
+
+				d.setServiceName(o.ServiceName)
 			} else {
-				// There's no alm examples in the CSV's definition
-				return fmt.Errorf("Could not find alm-examples in operator's definition.\nPlease provide a file containing yaml specification to start the %s service from %s operator", o.CustomResource, o.ServiceName)
+				o.ServiceName, err = d.getServiceNameFromCRD()
+				if err != nil {
+					return err
+				}
+			}
+
+			// CRD is valid. We can use it further to create a service from it.
+			o.CustomResourceDefinition = d.OriginalCRD
+
+			return nil
+		} else if o.CustomResource != "" {
+			// make sure that CSV of the specified ServiceType exists
+			csv, err := o.KClient.GetClusterServiceVersion(o.ServiceType)
+			if err != nil {
+				// error only occurs when OperatorHub is not installed.
+				// k8s does't have it installed by default but OCP does
+				return err
 			}
 
-			almExample, err := getAlmExample(almExamples, o.CustomResource, o.ServiceType)
+			almExample, err := svc.GetAlmExample(csv, o.CustomResource, o.ServiceType)
 			if err != nil {
 				return err
 			}
-			o.CustomResourceDefinition = almExample
-			o.group, o.version = groupVersionALMExample(almExample)
-			o.resource = resourceFromCSV(csv, o.CustomResource)
-			o.ServiceName, err = serviceNameFromCRD(o.CustomResourceDefinition, o.ServiceName)
-			return err
-		} else {
-			// prevent user from executing `odo service create <operator-name>`
-			// because the correct way is to execute `odo service
-			// <operator-name> --crd <crd-name>`
-			csvs, err := o.KClient.GetClusterServiceVersionList()
+
+			d.OriginalCRD = almExample
+
+			o.group, o.version, o.resource, err = svc.GetGVRFromOperator(csv, o.CustomResource)
 			if err != nil {
 				return err
 			}
 
-			for _, csv := range csvs.Items {
-				if csv.Name == o.ServiceType {
-					// this is satisfied if user has specified operator but not
-					// a CRD name
-					return errors.New("Please specify service name along with the operator name")
+			if o.ServiceName != "" && !o.DryRun {
+				// First check if service with provided name already exists
+				svcFullName := strings.Join([]string{o.CustomResource, o.ServiceName}, "/")
+				exists, _ := svc.OperatorSvcExists(o.KClient, svcFullName)
```

Comment:
Any reasons to inform the error here too?

Reply:
Same as https://github.com/openshift/odo/pull/3509/#discussion_r459906366.

Reply:
IMO we should never do this :( I agree with @mik-dass that we should check for error regardless